paperqa/agents/task.py

```diff
@@ -487,3 +493,251 @@ def __len__(self) -> int:
+        data_path: str,
+        num_questions: int | None = None,
+        settings: Settings | dict | None = None,
+        pairwise_eval_llm: LLMModel | str = CommonLLMNames.GPT_4O.value,
+    ):
+        if settings is None:
+            settings = Settings()
+        if isinstance(settings, dict):
+            settings = Settings(**settings)
+        self._settings = settings
+
+        if num_questions is not None:
+            self.data = pd.read_csv(data_path).head(num_questions)
+        else:
+            self.data = pd.read_csv(data_path)
+
+        self._rewards = {
+            "win": 1,
+            "tie": 0,
+            "lose": -1,
+        }
+
+        self.pai = pairwise_eval_llm
+
+    def get_new_env_by_idx(self, idx: int) -> GradablePaperQAEnvironment:
+        """Create a new environment instance for the given index."""
+        row = self.data.iloc[idx]
+
+        return LFRQAPairwiseEvalEnv(
+            qid=row.qid,
+            question=row.question,
+            human_answer=row.answer,
+            settings=self._settings,
+            rewards=self._rewards,
+            gt_doc_ids=row.gold_doc_ids.strip("[]").split(","),
+            pairwise_eval_llm=self.pai,
+        )
+
+    def compute_trajectory_metrics(
+        self, trajectories: "Sequence[Trajectory]"
+    ) -> dict[str, list[float]]:
+        metrics = super().compute_trajectory_metrics(trajectories)
+
+        # Add custom metrics similar to LitQATaskDataset
+        total_paper_count: list[float] = []
+        relevant_paper_count: list[float] = []
+        evidence_count: list[float] = []
+
+        for t in trajectories:
+            split_certainties = [
+                split_certainty
+                for split_certainty in (
+                    re.split(
+                        pattern=Complete.CERTAINTY_SPLIT_REGEX_PATTERN,
+                        string=obs.content,
+                        maxsplit=1,
+                    )
+                    for obs in t.steps[-1].next_observation
+                    if (
+                        isinstance(obs, ToolResponseMessage)
+                        and obs.name == Complete.TOOL_FN_NAME
+                    )
+                )
+                if len(split_certainty) >= 4  # noqa: PLR2004
+            ]
+
+            for i, metric_list in enumerate(
+                (total_paper_count, relevant_paper_count, evidence_count),
+                start=1,
+            ):
+                metric_list.append(
+                    sum(int(sa[i]) for sa in split_certainties) / len(split_certainties)
+                    if split_certainties
+                    else 0
+                )
+
+        return metrics | {
+            "total_paper_count": total_paper_count,
+            "relevant_paper_count": relevant_paper_count,
+            "evidence_count": evidence_count,
+            "paperqa_won": [
+                int(t.steps[-1].reward == self._rewards["win"]) for t in trajectories
+            ],
```

Comment:
Wdyt of notating who is the loser somehow, to be more intuitive?

`paperqa_beat_human` could be one idea